### PR TITLE
WIP: Add PassManagerSettings to GNA Pass Manager

### PR DESCRIPTION
**Pull request by aalbersk**
Add refactor changes made by @mdeisher that introduce PassManagerSettings and a test for Concat Quantization

### CODE REVIEW

File: inference-engine/src/gna_plugin/optimizer/gna_pass_manager.hpp [hunk 1]
```diff
@@ -149,19 +149,23 @@ DECL_PASS_BEFORE_COPY(UnrollTI);
 */
 DECL_PASS_BEFORE_COPY(RemoveConst);
 
+struct PassManagerSettings {
+    Policy policy;
+    /// @brief wether to run passes before copy
```
Comment (dorloff) [inline inference-engine/src/gna_plugin/optimizer/gna_pass_manager.hpp:154]: whether

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/concat_quantization.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2019 Intel Corporation
```
Comment (dorloff) [inline inference-engine/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/concat_quantization.cpp:1]: 2020

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/concat_quantization.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+//
+
+#include <vector>
+
+#include "subgraph_tests/concat_quantization.hpp"
+#include "common_test_utils/test_constants.hpp"
+
+using namespace LayerTestsDefinitions;
+const std::vector<InferenceEngine::Precision> netPrecisions = {
+        InferenceEngine::Precision::FP32
```
Comment (pavel-rodionov) [inline inference-engine/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/concat_quantization.cpp:12]: Are another precisions actually works?
If so could you please add them to test. If something does not work could you please create a ticket.

File: inference-engine/tests/functional/plugin/shared/src/subgraph_tests/concat_qunatization.cpp [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+void ConcatQuantization::SetUp() {
+    std::vector<size_t> inputShape;
+    InferenceEngine::Precision netPrecision;
+    std::tie(netPrecision, inputShape, targetDevice, configuration) = this->GetParam();
+    auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+
+    auto params = ngraph::builder::makeParams(ngPrc, {inputShape});
+
+    std::vector<size_t> outFormShapes1 = { 1, 5, 32 };
```
Comment (pavel-rodionov) [inline inference-engine/tests/functional/plugin/shared/src/subgraph_tests/concat_qunatization.cpp:46]: If internal shapes are fixed is there any sense to make network input shape as a parameter?

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.hpp [hunk 5]
```diff
@@ -34,6 +34,13 @@ typedef std::tuple<
         std::string                  // Target Device
 > basicParams;
 
+typedef std::tuple<
+    InferenceEngine::Precision,         // Network Precision
+    InferenceEngine::SizeVector,        // Input Shape
+    std::string,                        // Target Device
+    std::map<std::string, std::string>  //Configuration
+> basicParamsWithConfig;
```
Comment (pavel-rodionov) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.hpp:42]: @iefode @mikhail-treskin What do you think about this? Should we contribute additional configs to `LayerTestsUtils` ?

**Pull request merged**